tests/test_collections.py

```diff
@@ -0,0 +1,313 @@
+        except Exception:
+            for task in task_list:
+                task.cancel()
+
+        await asyncio.create_task(self.check_file(stage_file, wait=30, exists=False))
+        self.check_exposure_count("2020032700020", "LATISS/runs/quickLook", 0)
+        self.check_exposure_count("2022112200951", "LATISS/raw/all", 0)
+
+    async def testDoNoDeleteCollectionsTestCase(self):
+        fits_name = "AT_O_20221122_000951_R00_S00.fits.fz"
+        stage_file, butler_tasks = await self.load(fits_name, "collection_test_2.yaml")
+
+        self.check_exposure_count("2020032700020", "LATISS/runs/quickLook", 1)
+        self.check_exposure_count("2022112200951", "LATISS/raw/all", 1)
+
+        task_list = []
+        for butler_task in butler_tasks:
+            task = asyncio.create_task(butler_task())
+            task_list.append(task)
+        task_list.append(asyncio.create_task(self.interrupt_me()))
+        #
+        # kick off all the tasks, until one (the "interrupt_me" task)
+        # throws an exception
+        try:
+            await asyncio.gather(*task_list)
+        except Exception:
+            for task in task_list:
+                task.cancel()
+
+        await asyncio.create_task(self.check_file(stage_file, wait=30, exists=False))
+        self.check_exposure_count("2020032700020", "LATISS/runs/quickLook", 1)
+        self.check_exposure_count("2022112200951", "LATISS/raw/all", 0)
+
+    async def check_file(self, filename, wait=25, exists=True):
+        """Check that the existance of a file
+
+        Parameters
+        ----------
+        filename : `str`
+            name of the file to check
+        wait : `int`
+            seconds to wait until the file is checked
+        exists : `bool`
+            If True, file is checked that it exists
+            If False, file is checked that it doesn't exist
+        """
+
+        await asyncio.sleep(wait)
+        if exists:
+            self.assertTrue(os.path.exists(filename))
+            logging.info("file was there, as expected")
+        else:
+            self.assertFalse(os.path.exists(filename))
+            logging.info("file was not there, as expected")
+
+    def copy_to_test_location(self, fits_name):
+        # location of test file
+        test_dir = os.path.abspath(os.path.dirname(__file__))
+        data_file = os.path.join(test_dir, "data", fits_name)
+
+        # create a temp space, and copy the file there
+        sub_dir = tempfile.mkdtemp()
+        dest_file = os.path.join(sub_dir, fits_name)
+        logging.info("copying file {data_file=} to {dest_file=}")
+        shutil.copyfile(data_file, dest_file)
+        logging.info("file copied")
+        return dest_file
+
+    def check_exposure_count(self, exposure, collections, num_expected):
+        """check the number of exposures we expect to find
+
+        Parameters
+        ----------
+        exposure : `str`
+            the name of the exposure to check
```

Comment:
Parameters `collections` and `num_exptected` are not described. Please fix.

Reply:
Fixed